app/models/post.rb

```diff
@@ -0,0 +1,15 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  def update_posts_counter
+    counter = Post.where(author_id:).count
+    user = User.find_by(id: author_id)
+    user.update(posts_counter: counter)
+  end
```

Comment:
- [ ] Another way of doing this would be:
```rb
def update_post_counter
  author.increment!(:posts_counter)
end
```
Remember that we already set up associations between models.

Reply:
✅ Resolved.